addon/mixins/authenticated-route-mixin.js

```diff
@@ -33,12 +33,25 @@ export default Mixin.create({
   */
   session: service('session'),
 
+  _cookies: computed(function() {
+    return getOwner(this).lookup('service:cookies');
+  }),
+
   _isFastBoot: computed(function() {
     const fastboot = getOwner(this).lookup('service:fastboot');
 
     return fastboot ? fastboot.get('isFastBoot') : false;
   }),
 
+  _secureCookies: computed(function() {
+    if (this.get('_isFastBoot')) {
+      const fastboot = getOwner(this).lookup('service:fastboot');
+      return fastboot.get('request.protocol') === 'https';
+    }
+
+    return window.location.protocol === 'https:';
+  }).volatile(),
```

Comment:
We should move this into the `sessionAuthenticated` method so that we don't add any unnecessary properties via the mixin. Also this will only ever be used when running in FastBoot so `window.location.protocol` can be ignored.